src/dataset/create_dataset.py

```diff
@@ -269,15 +171,16 @@ def get_embeddings_at_index(texts: str, batch_idx: int, batch_size: int = 200):
         rate_limit = 3500 / 60  # Maximum embeddings per second
 
         start = time.time()
-        self.embeddings = np.zeros((len(self.embedding_strings), LEN_EMBEDDINGS))
+        embedding_strings = [chunk_data['text'] for chunk_data in self.metadata]
+        self.embeddings = np.zeros((len(embedding_strings), LEN_EMBEDDINGS))
 
         with concurrent.futures.ThreadPoolExecutor() as executor:
             futures = [executor.submit(
                 get_embeddings_at_index, 
-                self.embedding_strings[batch_idx:batch_idx+batch_size], 
+                embedding_strings[batch_idx:batch_idx+batch_size], 
                 batch_idx,
-                len(self.embedding_strings[batch_idx:batch_idx+batch_size])
-            ) for batch_idx in range(0, len(self.embedding_strings), batch_size)]
+                len(embedding_strings[batch_idx:batch_idx+batch_size])
+            ) for batch_idx in range(0, len(embedding_strings), batch_size)]
```

Comment:
how about having a chunker function that receives a list (or even just an iterable) and splits it up into chunks of n items long? It would make this a lot clearer